assets/dev/js/editor/utils/helpers.js

```diff
@@ -418,63 +418,39 @@ module.exports = {
 	},
 
 	isActiveControl: function( controlModel, values ) {
-		let condition,
-			conditions;
-
-		// TODO: Better way to get this?
-		if ( _.isFunction( controlModel.get ) ) {
-			condition = controlModel.get( 'condition' );
-			conditions = controlModel.get( 'conditions' );
-		} else {
-			condition = controlModel.condition;
-			conditions = controlModel.conditions;
-		}
+		const condition = controlModel.condition || controlModel.get?.( 'condition' );
+		let conditions = controlModel.conditions || controlModel.get?.( 'conditions' );
+
+		// If there is a simple condition, convert it to a complex condition.
+		if ( condition ) {
+			const terms = [];
+
+			Object.entries( condition ).forEach( ( [ conditionName, conditionValue ] ) => {
+				const conditionNameParts = conditionName.match( /(\w+(?:\[\w+])?)?(!?)$/i ),
+					conditionRealName = conditionNameParts[ 1 ],
+					isNegativeCondition = !! conditionNameParts[ 2 ];
+				let operator;
+
+				if ( Array.isArray( conditionValue ) && conditionValue.length ) {
+					operator = isNegativeCondition ? '!contains' : 'contains';
+				} else if ( isNegativeCondition ) {
+					operator = '!==';
+				}
 
-		// Multiple conditions with relations.
-		if ( conditions && ! elementor.conditions.check( conditions, values ) ) {
-			return false;
-		}
+				terms.push( {
+					name: conditionRealName,
+					operator: operator,
+					value: conditionValue,
+				} );
+			} );
 
-		if ( _.isEmpty( condition ) ) {
-			return true;
+			conditions = {
+				relation: 'and',
+				terms: conditions ? terms.concat( conditions ) : terms,
+			};
 		}
 
-		var hasFields = _.filter( condition, function( conditionValue, conditionName ) {
-			var conditionNameParts = conditionName.match( /([a-z_\-0-9]+)(?:\[([a-z_]+)])?(!?)$/i ),
-				conditionRealName = conditionNameParts[ 1 ],
-				conditionSubKey = conditionNameParts[ 2 ],
-				isNegativeCondition = !! conditionNameParts[ 3 ],
-				controlValue = values[ conditionRealName ];
-
-			if ( values.__dynamic__ && values.__dynamic__[ conditionRealName ] ) {
-				controlValue = values.__dynamic__[ conditionRealName ];
-			}
-
-			if ( undefined === controlValue ) {
-				return true;
-			}
-
-			if ( conditionSubKey && 'object' === typeof controlValue ) {
-				controlValue = controlValue[ conditionSubKey ];
-			}
```

Comment:
Where is the replacement for these checks?

Reply:
> Where is the replacement for these checks?

The `conditions` mechanism is a duplication of the `condition`, so it's already included all the checks required, I just needed to convert the `condition` to `conditions`. I deleted the duplication (the old version of `condition`).